finagle-core/src/main/scala/com/twitter/finagle/dispatch/ServerDispatcher.scala

```diff
@@ -18,28 +47,14 @@ object GenSerialServerDispatcher {
  * [[com.twitter.finagle.dispatch.SerialServerDispatcher SerialServerDispatcher]],
  * allowing the implementor to furnish custom dispatchers & handlers.
  */
-abstract class GenSerialServerDispatcher[Req, Rep, In, Out](trans: Transport[In, Out])
-    extends Closable {
+abstract class GenSerialServerDispatcher[Req, Rep, In, Out](trans: AnnotatedTransport[In, Out],
+  underTest: Boolean = false) extends ServerDispatcher[Req, Rep, In, Out](trans) {
```

Comment:
I personally prefer an Optional Context, but wanted to avoid introducing new types primarily because I wasn't sure how this was handled else where in the project.